dotnet/src/Planners/Planners.Core/Handlebars/ParameterType.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Planners.Handlebars;
+
+internal class Property
```

Comment:
'Property` is pretty generic, could this be called `HandlebarsProperty`? Similar for ParameterType.

Reply:
I changed it to "NestedProperty" for now but I'm going to remove it with the work to integrate your type support changes altogether